### PR TITLE
Refactor server to its own package in utils/server #261

### LINKED ISSUE #261: move server to its own package in utils/server
The `grpc_server.go` is getting super large. Let's move `server.go`, `grpc_server.go` and any other related file to another package (`utils/server`)

### PR DESCRIPTION
**Pull request by Inconnu08**
- Resolves #261

Additional minor changes
- Adds empty slice declaration via literals
- Refactors shadows vars
- Fixes typos

### CODE REVIEW

**Comment by YourTechBud** [comment]
Hey. Seems like youve done a lot of refactoring which is getting too difficult for me to check. A lot of the type name comments and error messages have been made lowercase, etc. I'll have to reject this change.

**Comment by Inconnu08** [comment]
@YourTechBud oh understandable, I could have refactored them back.

File: main.go [hunk 1]
```diff
@@ -98,13 +100,13 @@ func actionRun(c *cli.Context) error {
...
 	// Project and env cannot be changed once space cloud has started
-	s := initServer(isProd)
+	s := server.InitServer(isProd)
 
 	if !disableNats {
 		// TODO read nats config from the yaml file if it exists
 		if seeds != "" {
 			array := strings.Split(seeds, ",")
-			urls := []*url.URL{}
+			var urls []*url.URL
```
[the remark this answers is not in the document]
Comment (Inconnu08) [reply main.go]: This is creating a URL array

[the remark this answers is not in the document]
Comment (YourTechBud) [reply main.go]: No. Your code declaring an array, it isn't initialising an empty one

[the remark this answers is not in the document]
Comment (Inconnu08) [reply main.go]: I see

File: modules/realtime/realtime.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (m *Module) SetConfig(project string, conf *config.Realtime) error {
 
 	// Conect and create a new nats client
 	if conf.Broker != utils.Nats {
-		return errors.New("Realtime Error: Broker is not supported")
+		return errors.New("realtime Error: Broker is not supported")
```
[the remark this answers is not in the document]
Comment (YourTechBud) [reply modules/realtime/realtime.go]: Missed this change

File: utils/server/grpc.go [hunk 3]
```diff
@@ -162,22 +158,21 @@ func (s *server) Delete(ctx context.Context, in *pb.DeleteRequest) (*pb.Response
...
-	return &pb.Response{Status: 200}, nil
+	return &pb.Response{Status: http.StatusOK}, nil
 }
 
-func (s *server) Aggregate(ctx context.Context, in *pb.AggregateRequest) (*pb.Response, error) {
-
+func (s *Server) Aggregate(ctx context.Context, in *pb.AggregateRequest) (*pb.Response, error) {
 	req := model.AggregateRequest{}
-	temp := []map[string]interface{}{}
+	var temp []map[string]interface{}
```
Comment (YourTechBud) [inline utils/server/grpc.go]: You are just declaring the array here not initializing

File: utils/server/grpc.go [hunk 4]
```diff
@@ -191,27 +186,26 @@ func (s *server) Aggregate(ctx context.Context, in *pb.AggregateRequest) (*pb.Re
...
-
+func (s *Server) Batch(ctx context.Context, in *pb.BatchRequest) (*pb.Response, error) {
 	type msg struct {
 		id, col string
 	}
 
 	msgIDs := make([]*msg, len(in.Batchrequest))
 
-	allRequests := []model.AllRequest{}
+	var allRequests []model.AllRequest
```
Comment (YourTechBud) [inline utils/server/grpc.go]: You are just declaring the array here not initializing

File: utils/server/grpc.go [hunk 5]
```diff
@@ -352,69 +346,68 @@ func (s *server) Call(ctx context.Context, in *pb.FunctionsRequest) (*pb.Respons
 	return &pb.Response{Result: data, Status: 200}, nil
 }
 
-func (s *server) Service(stream pb.SpaceCloud_ServiceServer) error {
-	client := client.CreateGRPCServiceClient(stream)
-	defer s.functions.UnregisterService(client.ClientID())
-	defer client.Close()
-	go client.RoutineWrite()
+func (s *Server) Service(stream pb.SpaceCloud_ServiceServer) error {
+	gsClient := client.CreateGRPCServiceClient(stream)
```
Comment (YourTechBud) [inline utils/server/grpc.go]: Instead of using gsClient, lets just keep the variable name `c`?

File: utils/server/metrics.go [hunk 6]
```diff
@@ -169,15 +171,15 @@ func getProjectInfo(config *config.Modules) map[string]interface{} {
 	}
 
 	if config.Crud != nil {
-		crud := []string{}
+		var crud []string
```
Comment (YourTechBud) [inline utils/server/metrics.go]: You are just declaring the array not initializing it

File: utils/server/metrics.go [hunk 7]
```diff
@@ -169,15 +171,15 @@ func getProjectInfo(config *config.Modules) map[string]interface{} {
...
 		for k := range config.Crud {
 			crud = append(crud, k)
 		}
 		project["crud"] = crud
 	}
 
 	if config.Auth != nil {
-		auth := []string{}
+		var auth []string
```
Comment (YourTechBud) [inline utils/server/metrics.go]: You are just declaring the array not initializing it

File: utils/server/server.go [hunk 8]
```diff
@@ -112,25 +113,25 @@ func (s *server) loadConfig(config *config.Project) error {
...
 
-func (s *server) initGRPCServer(port string) {
+func (s *Server) initGRPCServer(port string) {
 	lis, err := net.Listen("tcp", ":"+port)
 	if err != nil {
 		log.Fatal("Failed to listen:", err)
 	}
 
-	options := []grpc.ServerOption{}
+	var options []grpc.ServerOption
```
Comment (YourTechBud) [inline utils/server/server.go]: You are just declaring the array not initializing it

File: utils/server/websocket.go [hunk 9]
```diff
@@ -18,45 +18,45 @@ var upgrader = websocket.Upgrader{
...
 		c, err := upgrader.Upgrade(w, r, nil)
 		if err != nil {
 			log.Println("upgrade:", err)
 			return
 		}
 
-		client := client.CreateWebsocketClient(c)
-		defer s.realtime.RemoveClient(client.ClientID())
-		defer s.functions.UnregisterService(client.ClientID())
+		wsClient := client.CreateWebsocketClient(c)
```
Comment (YourTechBud) [inline utils/server/websocket.go]: Lets just rename the client to `c`?

Comment (Inconnu08) [reply utils/server/websocket.go]: Alright but I had to rename the already c upgrader to u

File: utils/server/metrics.go [hunk 10]
```diff
@@ -67,7 +69,7 @@ func (t *transport) update(ctx context.Context, meta *proto.Meta, op string, fin
 
 // Init initialises a new transport
 func newTransport(host, port string, sslEnabled bool) (*transport, error) {
-	dialOptions := []grpc.DialOption{}
+	var dialOptions []grpc.DialOption
```
Comment (YourTechBud) [inline utils/server/metrics.go]: You are only declaring the array, not initialising it

File: utils/server/websocket.go [hunk 11]
```diff
@@ -18,45 +18,45 @@ var upgrader = websocket.Upgrader{
 	},
 }
 
-func (s *server) handleWebsocket() http.HandlerFunc {
+func (s *Server) handleWebsocket() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		c, err := upgrader.Upgrade(w, r, nil)
+		u, err := upgrader.Upgrade(w, r, nil)
```
Comment (YourTechBud) [inline utils/server/websocket.go]: lets rename `u` to `socket`

**Pull request merged**